### PR TITLE
Added comment about cluster API vs cloud providers to readme

### LINKED ISSUE #939: Updated README to clarify the difference between Cluster API and Kubernetes Cloud Provider
/kind bug

The Kubernetes also have bunch of cloud providers at https://github.com/kubernetes/kubernetes/tree/master/pkg/cloudprovider , I think we need to clarify the difference between those two providers, comments? @detiber @vincepri

### PR DESCRIPTION
**Pull request by liztio**
**What this PR does / why we need it**:
Adds a brief note to the README about cluster API vs Cloud Providers

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #939 

**Special notes for your reviewer**:

If this needs more detail let me know!

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,6 +14,16 @@ API design. Because of this, all of the prototype code is rapidly changing.
...
 
+### How does Cluster API compare to [Cloud Providers](https://kubernetes.io/docs/concepts/cluster-administration/cloud-providers/)?
+
+Cloud Providers and the Cluster API work in concert to provide a rich Kubernetes experience in cloud environments.
+The Cluster API initializes new nodes and clusters using available [providers](#Provider-Implementations).
+Running clusters can then use Cloud Providers to provision support infrastructure like 
+[load balancers](https://kubernetes.io/docs/tasks/access-application-cluster/create-external-load-balancer/)
+and [persistent volumes](https://kubernetes.io/docs/concepts/storage/persistent-volumes/).
+
+
```
Comment (gyliu513) [inline README.md]: Keeping one blank line is good enough.

File: README.md [hunk 2]
```diff
@@ -14,6 +14,16 @@ API design. Because of this, all of the prototype code is rapidly changing.
 
 Learn more about the project's [scope, objectives, goals and requirements](./docs/scope-and-objectives.md), [feature proposals](./docs/proposals/) and [reference use cases](./docs/staging-use-cases.md).
 
+### How does Cluster API compare to [Cloud Providers](https://kubernetes.io/docs/concepts/cluster-administration/cloud-providers/)?
```
Comment (gyliu513) [inline README.md]: How about s/Cloud Providers/Kubernetes Cloud Providers

**Comment by jichenjc** [comment]
looks great, just need fix the nits

**Comment by liztio** [comment]
@jichenjc updated

**Pull request merged**